File: Java/pattern/Decorator/src/com/pattern/Decorator.java
package com.pattern;

public class Decorator implements IComponent {

	private IComponent m_component;
	
	public Decorator(IComponent m_component) {
		super();
		this.m_component = m_component;
	}

	@Override
	public void showName() {
		m_component.showName();
	}

}

File: cplusplus/design_pattern/DecoratorProj/Decorator.cpp
#include "stdafx.h"
#include "Decorator.h"


Decorator::Decorator()
{
}

Decorator::Decorator(std::shared_ptr<IComponent> component) : m_component(component)
{
}


Decorator::~Decorator()
{
}

void Decorator::showName()
{
    m_component->showName();
}
